util/fipstools/acvp/acvptool/subprocess/kdf-components.go

```diff
@@ -55,134 +65,282 @@ type kdfCompTestGroupResponse struct {
 }
 
 type kdfCompTestResponse struct {
-	ID              uint64 `json:"tcId"`
-	MasterSecretHex string `json:"masterSecret"`
-	KeyBlockHex     string `json:"keyBlock"`
+	ID uint64 `json:"tcId"`
+	// Below values are unique to TLS Test Groups
+	MasterSecretHex string `json:"masterSecret,omitempty"`
+	KeyBlockHex     string `json:"keyBlock,omitempty"`
+	// Below values are unique to SSH Test Groups
+	InitialIvClientHex     string `json:"initialIvClient,omitempty"`
+	InitialIvServerHex     string `json:"initialIvServer,omitempty"`
+	EncryptionKeyClientHex string `json:"encryptionKeyClient,omitempty"`
+	EncryptionKeyServerHex string `json:"encryptionKeyServer,omitempty"`
+	IntegrityKeyClientHex  string `json:"integrityKeyClient,omitempty"`
+	IntegritykeyServerHex  string `json:"integrityKeyServer,omitempty"`
 }
 
 type kdfComp struct {
 	algo string
 }
 
-func (k *kdfComp) Process(vectorSet []byte, m Transactable) (interface{}, error) {
-	var parsed kdfCompVectorSet
-	if err := json.Unmarshal(vectorSet, &parsed); err != nil {
-		return nil, err
+var ivLenMap = map[string]uint32{
+	"AES-128": 16,
+	"AES-192": 16,
+	"AES-256": 16,
+	"TDES":    8,
+}
```

Comment:
These constants will never change but is there somewhere else we can pull them from that are already defined?

Reply:
Not that I'm aware of in AWS-LC. I've looked in [`sshkdf.h`](https://github.com/awslabs/aws-lc/blob/d998125d62c8bb009e18f4a95757c3c05f5709f9/include/openssl/sshkdf.h#L43), Chris' PR for [SSHKDF](https://github.com/awslabs/aws-lc/pull/656/files), and all over the golang files for the ACVP tool but didn't really find anything.

Reply:
`cipher.h` has `EVP_MAX_IV_LENGTH` = 16, but that's not particularly better, and there's nothing IV-related = 8.